dynamiqs/result.py

```diff
@@ -188,3 +188,49 @@ class MEResult(Result):
         [Batching simulations](../../documentation/basics/batching-simulations.md)
         tutorial for more details.
     """
+
+
+class PropagatorResult(Result):
+    r"""Result of the Schrödinger equation integration to obtain the propagator.
+
+    Attributes:
+        propagator _(array of shape (..., ntsave, n, n))_: Saved propagators.
+        infos _(PyTree or None)_: Solver-dependent information on the resolution.
+        tsave _(array of shape (ntsave,))_: Times for which results were saved.
+        solver _(Solver)_: Solver used.
+        gradient _(Gradient)_: Gradient used.
+        options _(Options)_: Options used.
+
+    Note-: Result of running multiple simulations concurrently
+        The resulting propagators are batched according to the
+        leading dimensions of the Hamiltonian `H`. In this case, there
+        is no sense of flat batching because the only object to batch over is
+        the Hamiltonian. Therefore setting `cartesian_batching=False` will throw an error.
+        This situation is to be contrasted with [`dq.sesolve`][dynamiqs.sesolve],
+        where the initial states are explicitly specified and one can have flat batching over
+        the Hamiltonian and initial states.
+
+        See the
+        [Batching simulations](../../documentation/basics/batching-simulations.md)
+        tutorial for more details.
+
+    """
+
+    @property
+    def propagator(self) -> Array:
+        return self._saved.ysave
+
+    @property
+    def states(self) -> Array:
+        raise NotImplementedError("states is not an attribute of PropagatorResult."
+                                  "Perhaps you meant to ask for the attribute propagator?")
```

Comment:
Nit:
```suggestion
                                  "The propagators can be accessed through the `.propagators` attribute.")
```
(also I'm not sure, but I believe it's not black formatted)